cmd/config/set.go

```diff
@@ -195,12 +209,21 @@ func configSetContext(cmd *cobra.Command, args []string) {
 		if err != nil {
 			log.Fatal(err.Error())
 		}
+		if !flags.Changed("patch") {
+			URLClearConditions(ctxPtr)
+		}
 		ctxPtr.URL = cleanedUrl
 	}
 	if flags.Changed("tenant") {
+		if ctxPtr.AuthMethod != "jwt" && !flags.Changed("patch") {
+			log.Fatalf("The tenant cannot be set for jwt. To override this, use the --patch flag")
+		}
 		ctxPtr.Tenant, _ = flags.GetString("tenant")
 	}
 	if flags.Changed("token") {
+		if ctxPtr.AuthMethod != "jwt" && !flags.Changed("patch") {
+			log.Fatalf("The tenant cannot be set for jwt. To override this, use the --patch flag")
```

Comment:
see note above. Token can be explicitly set ONLY for the JWT mode, so the message seems incorrect